core/metacling/src/TClingMethodInfo.cxx

```diff
@@ -53,117 +55,257 @@ compiler, not CINT.
 #include "llvm/Support/Casting.h"
 #include "llvm/Support/raw_ostream.h"
 
+#include <algorithm>
 #include <string>
 
 using namespace clang;
 
-TClingMethodInfo::TClingMethodInfo(const TClingMethodInfo &rhs) :
-   TClingDeclInfo(rhs),
-   fInterp(rhs.fInterp),
-   fContexts(rhs.fContexts),
-   fFirstTime(rhs.fFirstTime),
-   fContextIdx(rhs.fContextIdx),
-   fIter(rhs.fIter),
-   fTitle(rhs.fTitle),
-   fTemplateSpec(rhs.fTemplateSpec),
-   fDefDataSpecFuns(rhs.fDefDataSpecFuns),
-   fDefDataSpecFunIter(fDefDataSpecFuns.begin() + (rhs.fDefDataSpecFunIter - rhs.fDefDataSpecFuns.begin()))
+TClingCXXRecMethIter::SpecFuncIter::SpecFuncIter(cling::Interpreter *interp, clang::DeclContext *DC,
+                                                 llvm::SmallVectorImpl<clang::Decl*> &&specFuncs)
 {
-}
+   auto *CXXRD = llvm::dyn_cast<CXXRecordDecl>(DC);
+   if (!CXXRD)
+      return;
 
+   // Could trigger deserialization of decls.
+   cling::Interpreter::PushTransactionRAII RAII(interp);
 
-TClingMethodInfo& TClingMethodInfo::operator=(const TClingMethodInfo &rhs) {
-   if (this == &rhs)
-      return *this;
+   auto emplaceSpecFunIfNeeded = [&](clang::Decl *D) {
+      if (!D)
+         return; // Handle "structor not found" case.
 
-   this->TClingDeclInfo::operator=(rhs);
-   fInterp = rhs.fInterp;
-   fContexts = rhs.fContexts;
-   fFirstTime = rhs.fFirstTime;
-   fContextIdx = rhs.fContextIdx;
-   fIter = rhs.fIter;
-   fTitle = rhs.fTitle;
-   fTemplateSpec = rhs.fTemplateSpec;
-   fDefDataSpecFuns = rhs.fDefDataSpecFuns;
-   fDefDataSpecFunIter = fDefDataSpecFuns.begin() + (rhs.fDefDataSpecFunIter - rhs.fDefDataSpecFuns.begin());
+      if (std::find(CXXRD->decls_begin(), CXXRD->decls_end(), D) == CXXRD->decls_end()) {
+         fDefDataSpecFuns.emplace_back(llvm::dyn_cast<clang::FunctionDecl>(D));
+      }
+   };
 
-   return *this;
+   for (auto SpecFunc: specFuncs)
+      emplaceSpecFunIfNeeded(SpecFunc);
 }
 
+bool TClingCXXRecMethIter::Skip(const clang::Decl* D) const
```

Comment:
Skip is an action but the function return Yes/No.  
```suggestion
bool TClingCXXRecMethIter::ShouldSkip(const clang::Decl* D) const
```
would result in:
```
   If (ShouldSkip(decl))
     continue;
```

Reply:
Thanks, fixed.